app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -1172,8 +1172,14 @@ void displayMessage(String message, boolean bad, boolean suppressToast) {
     }
 
     public static boolean isDemoUser() {
-        User u = CommCareApplication._().getSession().getLoggedInUser();
-        return (User.TYPE_DEMO.equals(u.getUserType()));
+        try {
+            User u = CommCareApplication._().getSession().getLoggedInUser();
+            return (User.TYPE_DEMO.equals(u.getUserType()));
+        } catch (SessionUnavailableException e) {
+            // Default to a normal user: this should only happen if session
```

Comment:
But you're right, it is good to be cautious about this default value! I just checked and the uses of `isDemoeUser` toggle warning dialogs and enable the settings menu. If those don't or do show up when they shouldn't we should be fine.